### ISSUE TITLE
[BUG] (BABYSTEP + OFFSET_RANGE >9 ) -->float42_52 1/100 scaling display wrong value

### ISSUE THREAD

**Issue by studiodyne** [MarlinFirmware/Marlin#17230]
With
BABYSTEP_ZPROBE_OFFSET  & OFFSET_RANGE >9

Babystep display have bad floating point position 
float42-52 display type is wrong

**Comment by Kaween-prog** [comment]
Can confirm issue in 2.0.5.2, was ok in 2.0.5.1

Hardware : SKR1.3 + TMC2209 steppers 
Used LCD : 12864 RRDFGDC 
Delta Printer.
Using #define BABYSTEP_ZPROBE_OFFSET  active in configuration_adv.h

Firmware 2.0.5.1 
BABYSTEP_ZPROBE_OFFSET   :  value set to -16.20 (set by #define NOZZLE_TO_PROBE_OFFSET)
Expected result : Z offset showing -16.20 on LCD menu and correct function
Result : Z offset indeed showing -16.20   (correct in function and on LCD display)

Firmware 2.0.5.2 
BABYSTEP_ZPROBE_OFFSET   :  value set to -16.20  (set by #define NOZZLE_TO_PROBE_OFFSET)
Expected result : Z offset showing -16.20 on LCD menu and correct function
Result : Z offset showing value 10x smaller (decimal point shifted) at -1.620  (display showing -1.620 instead of -16.20 and nozzle crashses into bed)

Initialising eeprom, restoring failsafe +  store  : situation does not change. Value remains wrong.

Rollback to 2.0.5.1 : no issues.

**Comment by Taomyn** [comment]
I'm seeing the same thing between 2.0.5.1 and 2.0.5.2 and does seem to be a display issue as the value is stored correctly it's just shown as a number 10x smaller. Doesn't seem to affect anything as far as I can tell.

**Comment by studiodyne** [comment]
In menu.h , it's a just a /100 scale
```
#if HAS_BED_PROBE
  #if Z_PROBE_OFFSET_RANGE_MIN >= -9 && Z_PROBE_OFFSET_RANGE_MAX <= 9
    #define LCD_Z_OFFSET_TYPE float43    // Values from -9.000 to +9.000
  #else
    #define LCD_Z_OFFSET_TYPE float42_52 // Values from -99.99 to 99.99
  #endif
#endif
```
I suggest to change float42_52 by a float53 or anything without scaling

**Comment by Roxy-3D** [comment]
I recently made a change to add an extra decimal place of precision to that number on the display.
And the space for that extra digit came from the left side of the decimal point.
It may be we need a new, higher precision edit menu type.   Maybe 6.3 ?    

Does reverting this change fix your display?  https://github.com/MarlinFirmware/Marlin/commit/9f85165101008a87f37cb935591522e45910e848

If so...   I think we better add a new edit menu type!

**Comment by studiodyne** [comment]
the best is to create a new float42_52 displaying but with no scale
Create it 👍

**Comment by Roxy-3D** [comment]
So what do you think this float42_52 is exactly?    Why is it some times 4 and some times 5 ?
And 2 decimal places is not enough for precision machines and settings.   We probably need a 3 decimal place version of what ever we decide we need.

**Comment by ropch** [comment]
Is this behavior due to a typo in the multiplier within the new ftostr53sign function?

The following change corrected the display on my machine:
![image](https://user-images.githubusercontent.com/23707453/77715706-b1702500-6fb2-11ea-967b-c164509810e3.png)

**Comment by Roxy-3D** [comment]
@ropch   Is that change still working for you?   Is it well enough tested that we can confidently change that line in the repository?   (I'll zap it if you know it fixes the problem.)

**Comment by ropch** [comment]
Yes, I have the change in place and it is working for me.

I noticed the problem right after moving from 2.0.5.1 to 2.0.5.2, upon setting my z-offset.

Although I'm new to Marlin, looking over commits between 2.0.5.1 and 2.0.5.2, I saw that my configuration utilized the new ftostr53sign function.  The multiplier in that function appeared to be incorrect, and changing it (to * 10000) fixed the problem for me.

My fixed build has been working correctly since a few hours after 2.0.5.2 was tagged.  So, it gets the "It works on my machine" seal of approval from me.  ;)
